directory/cli/src/hostgroup.py

```diff
@@ -0,0 +1,131 @@
+import click
+from collections import OrderedDict
+
+import list_command
+from list_command import field_with_same_name    	
+import ipa_wrapper_command
+import ipa_utils
+from exceptions import IpaRunError
+
+HOSTGROUP_LIST_FIELD_CONFIGS = OrderedDict([
+	('Host group name', field_with_same_name),
+	('Description', field_with_same_name),
+	('HGID', field_with_same_name),
+])
+
+HOSTGROUP_SHOW_FIELD_CONFIGS = HOSTGROUP_LIST_FIELD_CONFIGS.copy()
+HOSTGROUP_SHOW_FIELD_CONFIGS.update([
+	('Member hosts', field_with_same_name),
+])
+
+HOSTGROUP_OPTIONS = {
+	'--desc':{'help': 'Host group description'}
+}
+# TODO enter blacklist hostgroups
+HOSTGROUP_BLACKLIST = []
+
+def add_commands(directory):
+
+	@directory.group(help='Manage host groups')
+	def hostgroup():
+		pass
+
+	@hostgroup.command(help='List all host groups')
+	def list():
+		list_command.do(
+			ipa_find_command='hostgroup-find',
+			field_configs=HOSTGROUP_LIST_FIELD_CONFIGS,
+			# TODO sort out the sort of the list results &
+			#sort_key='HGID',
+			#blacklist_key='Host group name',
+			blacklist_val_array=HOSTGROUP_BLACKLIST,
+		)
+	
+	@hostgroup.command(help='Show detailed information on one host group')
+	#this line gives the host group's name as an argument to the command call above
+	@click.argument('hostgroup_name')
+	def show(hostgroup_name):
+		_validate_blacklist_hostgroups(hostgroup_name)
+		## i don't understand this line
+		hostgroup_find_args = ['--hostgroup-name={}'.format(hostgroup_name)]
```

Comment:
This variable constructs the argument given to `ipa hostgroup-find` to locate the given host. The argument given to the show function (`hostgroup_name`) is substituted in the braces such that the variable will look like `--hostgroup-name=ipaservers` (if the userware command that was run was `hostgroup show ipaservers`)